src/windows/map.c

```diff
@@ -1349,6 +1349,12 @@ static void map_window_increase_map_size()
 	map_extend_boundary_surface();
 	window_map_init_map();
 	window_map_center_on_view_point();
+    // Recalculate fences
+	for (int y = 0; y < RCT2_GLOBAL(RCT2_ADDRESS_MAP_MAX_XY, uint16); y += 32) {
```

Comment:
It could be moved to a function in `src/world/map.c`, this will then be the only place where it's being called though. I didn't see any function doing this already.